### PR TITLE
Small gun changes

### PR DESCRIPTION
**Pull request by Gristlebee**
## About The Pull Request

Makes the combat shotgun bulky instead of huge, and the compact combat shotgun medium instead of bulky

Repaths the saber SMG to be a child of the skm carbine so the stock is able to fold and changes

Adds automatic fire to the TEC-9 and makes it normal sized.

## Why It's Good For The Game

The combat shotgun being huge is a holdover from station based server balance. It being bulky now will make it be able to fit in armor vests.  Moves the compact combat shotgun down to normal as a result since it's supposed to be the compact version.

The Saber and TEC-9 changes are so they're accurate to their descriptions of having a folding stock and having a fast rate of fire respectively.  Ups the TEC-9 to normal sized as a result since having an automatic pistol you can fit in your boot is probably not good.

## Changelog

:cl:

tweak: Saber SMG stock folds
tweak: TEC-9 Pistol is now automatic and normal sized
balance: Combat shotgun is now bulky, and the compact combat shotgun is normal.
/:cl:

### CODE REVIEW

**Review by rye-rice — APPROVED** [review]
Please be aware anything with the old gun pallete/ WS era zeta style is intentionally elft untouched as the proto SMG is unlikely to survive a planned NT/SL gun rework, and also be aware that i'm avoiding putting more things under skm_carbine as foldable stocks are meant to be attachments, eg you dont need to do weird subtyping like this.  

This approval is for the sprites and shotgun, I'm unsure about the changes i mentioned however.

**Comment by rye-rice** [comment]
Oh and the TEC-9 has a 0% chance of living post scarborgh PR, be aware of that as well

File: code/modules/projectiles/guns/ballistic/pistol.dm [hunk 1]
```diff
@@ -235,14 +235,18 @@ EMPTY_GUN_HELPER(automatic/pistol/commander/inteq)
...
 	show_magazine_on_sprite = TRUE
 	load_sound = 'sound/weapons/gun/pistol/mag_insert.ogg'
 	load_empty_sound = 'sound/weapons/gun/pistol/mag_insert.ogg'
 	eject_sound = 'sound/weapons/gun/pistol/mag_release.ogg'
 	eject_empty_sound = 'sound/weapons/gun/pistol/mag_release.ogg'
 
+/obj/item/gun/ballistic/automatic/pistol/tec9/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.3 SECONDS)
+
```
Comment (FalloutFalcon) [inline code/modules/projectiles/guns/ballistic/pistol.dm]: retlaw refactored adding automatic fire to weapons
use gun_firemods and fire_delay

**Review by FalloutFalcon — CHANGES_REQUESTED** [review]
looks good otherwise

File: code/modules/projectiles/guns/ballistic/smg.dm [hunk 2]
```diff
@@ -403,3 +392,20 @@ EMPTY_GUN_HELPER(automatic/smg/cm5)
 
 	unfolded_slowdown = 0.35
 	unfolded_wield_delay = 0.4 SECONDS
+
+
+/obj/item/gun/ballistic/automatic/smg/skm_carbine/inteq/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.13 SECONDS)
```
Comment (FalloutFalcon) [inline code/modules/projectiles/guns/ballistic/smg.dm]: Same situation as the tec9.
Use the refactored firing mode handling, Sorry I meant for both of them

**Review by FalloutFalcon — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic/smg.dm (hunk 2 above)
Comment (Gristlebee) [reply code/modules/projectiles/guns/ballistic/smg.dm]: Removed the automatic fire component and it's inheriting the needed behavior. Shooould be good if tests don't explode.

**Review by FalloutFalcon — APPROVED** [review]

**Review by FalloutFalcon — APPROVED** [review]

**Pull request merged**